spec/components/work_package_relations_tab/relations_mediator_spec.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe WorkPackageRelationsTab::RelationsMediator do
+  shared_let(:user) { create(:admin) }
+  shared_let(:project) { create(:project) }
+  shared_let(:work_package) { create(:work_package, project:) }
+
+  shared_let(:today) { Time.zone.today }
+
+  current_user { user }
+
+  subject(:mediator) { described_class.new(work_package:) }
+
+  describe "#relation_group", "follows" do
+    shared_let(:related_work_package1) { create(:work_package, subject: "WP #1", project:, due_date: today + 3.days) }
+    shared_let(:related_work_package2) { create(:work_package, subject: "WP #2", project:, due_date: today + 2.days) }
+    shared_let(:related_work_package3) { create(:work_package, subject: "WP #3", project:, due_date: today + 2.days) }
+    shared_let(:related_work_package4) { create(:work_package, subject: "WP #4", project:) }
```

Comment:
swapping the line with `related_work_package4` with the line `related_workPackage1` makes the test fail.

Reply:
This should be fixed - `RelationGroup#closest_relations` now excludes relations without dates.